browser/ui/webui/new_tab_page/brave_new_tab_message_handler_unittest.cc

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/new_tab_page/brave_new_tab_message_handler.h"
+
+#include "base/test/simple_test_clock.h"
+#include "base/time/time.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+TEST(BraveNewTabMessageHandlerTest, TalkPrompt) {
```

Comment:
Hmm, this test failed on my non-release local build on macOS and linux.
```
[ RUN      ] BraveNewTabMessageHandlerTest.TalkPrompt
../../brave/browser/ui/webui/new_tab_page/brave_new_tab_message_handler_unittest.cc:18: Failure
Expected equality of these values:
  BraveNewTabMessageHandler::CanPromptBraveTalk(clock->Now())
    Which is: false
  true
Stack trace:
0   brave_unit_tests                    0x000000010d813182 BraveNewTabMessageHandlerTest_TalkPrompt_Test::TestBody() + 658

[  FAILED  ] BraveNewTabMessageHandlerTest.TalkPrompt (9 ms)
[1/1] BraveNewTabMessageHandlerTest.TalkPrompt (9 ms)
1 test failed:
    BraveNewTabMessageHandlerTest.TalkPrompt (../../brave/browser/ui/webui/new_tab_page/brave_new_tab_message_handler_unittest.cc:12)
Tests took 0 seconds.
```

Reply:
With below change, this test is passed on my local.
```
diff --git a/browser/ui/webui/new_tab_page/brave_new_tab_message_handler_unittest.cc b/browser/ui/webui/new_tab_page/brave_new_tab_message_handler_unittest.cc
index 20bd237318..5d564eb03b 100644
--- a/browser/ui/webui/new_tab_page/brave_new_tab_message_handler_unittest.cc
+++ b/browser/ui/webui/new_tab_page/brave_new_tab_message_handler_unittest.cc
@@ -7,10 +7,12 @@

 #include "base/test/simple_test_clock.h"
 #include "base/time/time.h"
+#include "chrome/browser/first_run/first_run.h"
 #include "testing/gtest/include/gtest/gtest.h"

 TEST(BraveNewTabMessageHandlerTest, TalkPrompt) {
   auto* clock = new base::SimpleTestClock();
+  clock->SetNow(first_run::GetFirstRunSentinelCreationTime());
   EXPECT_EQ(BraveNewTabMessageHandler::CanPromptBraveTalk(clock->Now()), false);
   clock->Advance(base::TimeDelta::FromDays(2));
   EXPECT_EQ(BraveNewTabMessageHandler::CanPromptBraveTalk(clock->Now()), false);
```

Reply:
That's weird, it worked for me and all CI platforms. But happy to include your suggestion to prevent others having issues.

Reply:
I've pushed that change now.